doc/maintenance.md

```diff
@@ -146,6 +146,46 @@ supported version of Python. Minimum Python and NumPy version support should be
 adjusted upward on every major and minor release, but never on a patch release.
 
 
+## Backwards compatibility and deprecation policy
+
+PyGMT is still undergoing rapid developement. All of the API is subject to change
+until the v1.0.0 release.
+
+Basic policy for backwards compatibility:
+
+- Any incompatible changes should go through the deprecation process below.
+- Incompatible changes are allowed in major and minor releases, not in patch
+  releases.
+- Incompatible changes should be documented in the release notes.
+
+When making incompatible changes, we should follow the process:
+
+- Discuss whether the incompatible changes are necessary on GitHub.
+- Make the changes in a backwards compatible way, and raise a `FutureWarning`
+  warning for old usage. At least one test using the old usage should be added.
+- The warning message should clearly explain the changes and include the versions
+  in which the old usage is deprecated and is expected to be removed.
+- The `FutureWarning` warning should appear in 2-4 minor versions, depending on
```

Comment:
May want to wait for votes to come in on https://github.com/GenericMappingTools/pygmt/issues/1120#issuecomment-818377643 before committing.

```suggestion
- The `FutureWarning` warning should appear for 2-4 minor versions, depending on
```